tests/parameter_handler/parameter_handler_write_json.cc

```diff
@@ -59,8 +59,18 @@ main()
   prm.leave_subsection();
 
 
-  prm.print_parameters(deallog.get_file_stream(), ParameterHandler::JSON);
+  prm.print_parameters(deallog.get_file_stream(),
+                       ParameterHandler::JSON,
+                       true,
+                       true);
   deallog.get_file_stream() << std::endl;
 
+  prm.print_parameters(deallog.get_file_stream(),
+                       ParameterHandler::JSON,
+                       true,
+                       false);
```

Comment:
You can test my hypothesis that you're changing the state of the object by adding another `..., true)` output here and checking that the first and third outputs are the same :-)